### ISSUE TITLE
Automatic line wrapping in the fullscreen editor

### ISSUE THREAD

**Issue by patrikaxelsson** [punktniklas/NiKom#102]
Scenario:
You are writing a message and decide you need to change something say ten lines up which makes the line longer.

Currently, you first have to manually wrap/break the actual line to be able to enter the text and then you have to manually wrap all the following lines if you want the message to look tidy.

It would be very nice if the editor did the wrapping for you instead and it would save a lot of time for people like me who have a hard time writing a text without multiple edits.

**Comment by punktniklas** [comment]
I agree that it's a bit annoying that you most of the time need to "make room" on a line before you can add text to it. But I'm not sure what the best solution would be. Should it push the rest of the text in front of it and re-wrapping all the lines below as needed? For the entire text or just the current paragraph. (However a paragraph is defined.)
What about when you delete text, should it do the reverse and always pull text up from lines below as room becomes available?

There are probably cases where this is not desireable. E.g. when editing code snippets.

**Comment by patrikaxelsson** [comment]
I think it would be most useful if it pushed the text in front of it, rewrapping the text in the current paragraph to make room. And for delete it should pull up text from below lines of the current paragraph.

The definition of a paragraph would depend on how NiKom stores the text, I guess. For example if it was possible to define a paragraph as a very long line, ending with the first intentional line break, the definition would be clear.

If the definition what a paragraph is is not clear and the code has to make an educated guess, I can see that there could be occasional issues while editing code snippets. However, I think it would be a big convenience in most cases.

**Comment by adamel** [comment]
Reflow should definitely only happen within the current paragraph. I think paragraphs should be delimited by two newlines in a row, i.e.

This is the first paragraph.

This is the second paragraph.
Still still the second.

This is the third paragraph.

**Comment by adamel** [comment]
Newline followed by whitespace (another newline, space, tab) may be a good general rule for paragraph separators. That will also prevent reflow into indented code lines.
